daemon/daemon.go

```diff
@@ -387,12 +386,12 @@ func (d *Daemon) Start() error {
 }
 
 // HandleRestart implements overlord.RestartBehavior.
-func (d *Daemon) HandleRestart(t restart.RestartType) {
+func (d *Daemon) HandleRestart(t restart.RestartType, rebootInfo *boot.RebootInfo) {
```

Comment:
I think this needs to remember rebbotInfo in the daemon?

Reply:
Right, I've added that now.

Reply:
However, I am unsure on the `d.expectedRebootDidNotHappen == true`, as I think RebootInfo would not be available in that case.